File: transpiled_after/find-the-original-typed-string-ii/cpp/find-the-original-typed-string-ii_benchmark.py
# coding: utf-8
import time

# ========== PASSING PYTHON 2 SOLUTION (100% UNMODIFIED) ==========
# Time:  O(n + k^2)
# Space: O(n + k)

# dp, prefix sum
class Solution(object):
    def possibleStringCount(self, word, k):
        MOD = 10**9+7
        cnt = [0]
        for i in xrange(len(word)):
            cnt[-1] += 1
            if i+1 == len(word) or word[i+1] != word[i]:
                cnt.append(0)
        cnt.pop()
        result = reduce(lambda accu, x: accu*x%MOD, cnt, 1)
        if k <= len(cnt):
            return result
        dp = [0]*(k-len(cnt))
        dp[0] = 1
        for l in cnt:
            for i in xrange(len(dp)-1):
                dp[i+1] = (dp[i+1]+dp[i])%MOD
            for i in reversed(xrange(l, len(dp))):
                dp[i] = (dp[i]-dp[i-l])%MOD
        return reduce(lambda accu, x: (accu-x)%MOD, dp, result)

# ==================================================================

# ========== TEST HARNESS ==========
def main():
    iterations = 1000

    # Test cases extracted from C++ (convert C++ syntax to Python 2)
    test_cases = [
        ("", 1),
        ("a", 1),
        ("aaaaa", 3),
        ("abcde", 2),
        ("aaabbcdddde", 7),
        ("ababababab", 5),
        ("zzzzzyyyyyxxxxx", 8),
        ("q", 10),
        ("mississippi", 6),
        ("aabbccddeeffgg", 9)
    ]

    # Warm-up
    sol = Solution()
    for test in test_cases:
        if isinstance(test, tuple):
            sol.possibleStringCount(*test)
        else:
            sol.possibleStringCount(test)

    # Benchmark
    results = []
    start_time = time.time()

    for _ in xrange(iterations):
        sol = Solution()
        for test in test_cases:
            if isinstance(test, tuple):
                result = sol.possibleStringCount(*test)
            else:
                result = sol.possibleStringCount(test)
            results.append(result)

    elapsed_time = time.time() - start_time
    checksum = sum(hash(str(r)) for r in results) % (10**9 + 7)
    print("Checksum: {}, Total time: {:.6f}s".format(checksum, elapsed_time))

if __name__ == "__main__":
    main()

File: cpp_energy_test_final_feasible/find-the-original-typed-string-ii/find-the-original-typed-string-ii_benchmark.cpp
#include <iostream>
#include <vector>
#include <string>
#include <chrono>
#include <utility>

using namespace std;

class Solution {
public:
    int possibleStringCount(string word, int k) {
        const int MOD = 1000000007;
        vector<int> cnt;
        int current = 0;
        for (int i = 0; i < static_cast<int>(word.size()); ++i) {
            current += 1;
            if (i + 1 == static_cast<int>(word.size()) || word[i + 1] != word[i]) {
                cnt.push_back(current);
                current = 0;
            }
        }
        long long result = 1;
        for (int l : cnt) {
            result = (result * l) % MOD;
        }
        if (k <= static_cast<int>(cnt.size())) {
            return static_cast<int>(result);
        }
        int m = k - static_cast<int>(cnt.size());
        vector<long long> dp(m, 0);
        dp[0] = 1;
        for (int l : cnt) {
            for (int i = 0; i < m - 1; ++i) {
                dp[i + 1] = (dp[i + 1] + dp[i]) % MOD;
            }
            for (int i = m - 1; i >= l; --i) {
                dp[i] = (dp[i] - dp[i - l]) % MOD;
                if (dp[i] < 0) dp[i] += MOD;
            }
        }
        long long sumdp = 0;
        for (int i = 0; i < m; ++i) {
            sumdp = (sumdp + dp[i]) % MOD;
        }
        long long ans = (result - sumdp) % MOD;
        if (ans < 0) ans += MOD;
        return static_cast<int>(ans);
    }
};

int main() {
    vector<pair<string, int>> tests = {
        {"", 1},
        {"a", 1},
        {"aaaaa", 3},
        {"abcde", 2},
        {"aaabbcdddde", 7},
        {"ababababab", 5},
        {"zzzzzyyyyyxxxxx", 8},
        {"q", 10},
        {"mississippi", 6},
        {"aabbccddeeffgg", 9}
    };

    Solution sol;
    volatile unsigned long long checksum = 0;

    auto start = std::chrono::high_resolution_clock::now();

    const int iterations = 1000;
    for (int iter = 0; iter < iterations; ++iter) {
        for (const auto& t : tests) {
            checksum += sol.possibleStringCount(t.first, t.second);
        }
    }

    auto end = std::chrono::high_resolution_clock::now();
    auto elapsed_ms = std::chrono::duration_cast<std::chrono::milliseconds>(end - start).count();

    cout << "Checksum: " << checksum << "\n";
    cout << "Elapsed milliseconds: " << elapsed_ms << "\n";

    return 0;
}